LX/src/main/java/heronarts/lx/LXSnapshotActivePattern.java

```diff
@@ -0,0 +1,39 @@
+package heronarts.lx;
+
+import com.google.gson.JsonObject;
+
+public class LXSnapshotActivePattern extends LXComponent {
+
+  public LXChannel channel;
```

Comment:
These raw references are fine for a quick-and-dirty solution now. But for a long-term version they shouldn't be public and mutable.

You can take a look at how the Undo engine handles this stuff via the ComponentReference and ParameterReference classes. I will eventually bubble those up into a more public location in the API.